operator/pkg/ciliumendpointbatch/endpointbatch.go

```
@@ -0,0 +1,300 @@
+		clientV2:                            client.CiliumV2(),
+		clientV2a1:                          client.CiliumV2alpha1(),
+		reconciler:                          newReconciler(client.CiliumV2alpha1(), manager),
+		Manager:                             manager,
+		queue:                               rlQueue,
+		ciliumEndpointBatchStore:            cebStore,
+		ciliumEndpointBatchUpsertSyncPeriod: cebSyncPeriod,
+		ciliumEndpointBatchDeleteSyncPeriod: cebDeleteSyncPeriod,
+	}
+}
+
+// start the worker thread, reconciles the modified CEBs with api-server
+func (c *CiliumEndpointBatchController) Run(ces cache.Indexer, stopCh chan struct{}) {
+
+	log.Info("Bootstrap ceb controller")
+	defer utilruntime.HandleCrash()
+	defer c.queue.ShutDown()
+
+	// Cache CiliumEndpointStore Interface locally
+	c.ciliumEndpointStore = ces
+
+	// On operator warm boot, remove stale cep entries present in CEB
+	c.ciliumRemoveStaleCepEntries()
+
+	log.WithFields(logrus.Fields{
+		"Upsert syncperiod": c.ciliumEndpointBatchUpsertSyncPeriod,
+		"Delete syncperiod": c.ciliumEndpointBatchDeleteSyncPeriod,
+	}).Debug("Starting CEB controller reconciler.")
+
+	// TODO: multiple worker threads can run concurrently to reconcile with api-server
+	go wait.Until(c.worker, c.ciliumEndpointBatchUpsertSyncPeriod, stopCh)
+
+	go func() {
+		defer utilruntime.HandleCrash()
+	}()
+
+	<-stopCh
+
+	return
+}
+
+// Upon warm boot[restart], Iterate over all CEPs which we got from the api-server
+// and compare it with CEPs packed inside CEB.
+// If there are any stale CEPs present in CEBs, remove them from their CEB.
+func (c *CiliumEndpointBatchController) ciliumRemoveStaleCepEntries() {
+	log.Info("Remove stale CEP entries in CEB")
+
+	// Get all ceps from local datastore
+	staleCeps := c.Manager.getAllCepNames()
+
+	// Remove stale CEP entries present in CEB
+	for _, cepName := range staleCeps {
+		if _, exists, err := c.ciliumEndpointStore.GetByKey(cepName); err == nil && exists || err != nil {
```

Do we need to handle err here instead of treating it as "exists"?

---

Rather, i prefer check `err` and `exists`.

---

that's okay but why do we intentionally ignore error here? should at least throw a warning?

---

As of current implementation `GetByKey` never throws error, during review @varunmar raised a good point, what happens if `GetByKey` implementation changes and they throw error, in that case we don't want to `removecep` from cache, rather we ignore that error.

---

I see, can you add a comment stating we intentionally ignore the error and the reason?

---

I'll add a comment, I'm just saying we have too lengthy comments at many places, that unnecessarily raises eyebrow.